src/Orleans.Transactions/DistributedTM/TransactionAgent.cs

```diff
@@ -99,90 +89,118 @@ private async Task<TransactionalStatus> CommitReadOnlyTransaction(TransactionInf
             {
                 // wait for all responses
                 await Task.WhenAll(tasks);
+            }
+            catch (TimeoutException)
+            {
+                if (logger.IsEnabled(LogLevel.Debug))
+                    logger.Debug($"{stopwatch.Elapsed.TotalMilliseconds:f2} timeout {transactionInfo.TransactionId} on CommitReadOnly");
+                status = TransactionalStatus.ParticipantResponseTimeout;
+            }
+            catch (Exception ex)
+            {
+                if (logger.IsEnabled(LogLevel.Debug))
+                    logger.Debug($"{stopwatch.Elapsed.TotalMilliseconds:f2} failure {transactionInfo.TransactionId} CommitReadOnly");
+                this.logger.LogWarning(ex, "Unknown error while commiting readonly transaction {TransactionId}", transactionInfo.TransactionId);
+                status = TransactionalStatus.PresumedAbort;
+            }
 
-                // examine the return status
-                foreach (var s in tasks)
+            // examine the return status
+            if (status == TransactionalStatus.Ok) foreach (var s in tasks)
```

Comment:
This is pretty uncommon style in our codebase - suggest braces around the foreach block